server/confluence_config.go

```diff
@@ -28,14 +30,32 @@ func (p *Plugin) handleConfluenceConfig(w http.ResponseWriter, r *http.Request)
 
 	decoder := json.NewDecoder(r.Body)
 	submitRequest := &model.SubmitDialogRequest{}
+	response := &model.SubmitDialogResponse{}
 	if err := decoder.Decode(&submitRequest); err != nil {
 		p.API.LogError("Error decoding the submit dialog request.", "Error", err.Error())
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
+	configInstanceURL := submitRequest.Submission[configServerURL].(string)
+
+	if vErr := utils.IsValidURL(configInstanceURL); vErr != nil {
+		response.Error = ErrorInvalidURL
+
+		response, mErr := json.Marshal(response)
+		if mErr != nil {
+			p.API.LogError("Error in marshalling the URL.", "Error", mErr.Error())
+			http.Error(w, mErr.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		p.API.LogError("Error in validating the URL.", "Error", vErr.Error())
```

Comment:
shouldn't the message be different? This error will occur when the link is not valid, but this statement means that we were not able to validate the link itself.

Reply:
This error could also come if we are unable to validate the URL due to any number of reasons that's why it's better to log this statement and the main error will be logged after that.I am open to suggestions if you have a better idea

Reply:
this seems fine